src/uct/tcp/tcp.h

```diff
@@ -337,6 +346,8 @@ struct uct_tcp_ep {
                                                      * closed as soon as the EP is connected
                                                      * using the new fd */
     uct_tcp_ep_cm_id_t            cm_id;            /* EP connection mananger ID */
+    uint32_t                      last_acked_sn;    /* Last acked operation sequence number */
+    uint64_t                      put_cnt;          /* Number of PUT operations scheduled */
```

Comment:
on other handm making it 64but increases ep size. we can't really have 4G outstanding PUT operations, otheriwse sn would wrap around anyway, right?
so better return NO_RESOURCES if put_cnt >= INT32_MAX/2

Reply:
sorry, I don't see any differences to have `uint32_t put_cnt` and check `put_cnt == UINT32_MAX` instead of
```
so better return NO_RESOURCES if put_cnt >= INT32_MAX/2
```
Also, it will require checking `put_cnt` for other operations too. Then better to return back to what I suggested in https://github.com/openucx/ucx/pull/7140/commits/5a537850226e6c5ac7ee3a6d851d6104ba141540, i.e. having counter to simplify checking condition (it will be sing `if` condition instead of 3 ones)

Reply:
I think it makes sense to have 32-bit `put_cnt` and move it along with `last_acked_sn` under `uct_tcp_ep_ctx_t`. So, they will be inside union and `put_cnt` is valid for TX context, `last_acked_sn` - for RX context.